src/basic/fs-util.c

```diff
@@ -304,6 +317,19 @@ int fchmod_opath(int fd, mode_t m) {
         return 0;
 }
 
+int fchmodat_best(int dirfd, const char *path, mode_t mode, int flags) {
```

Comment:
Hmm, rethinking about this, is it really needed? `fchmod_opath` fundamentally does the same thing, and also handles empty `path` correctly.

Reply:
And this is broken if AT_EMPTY_PATH is used and we use the fallback path. So I don't really see much of value here.